meshtastic/node.py

```
@@ -998,27 +1058,99 @@ def setURL(self, url: str, addOnly: bool = False) -> None:
-                    disabled_channel.role = channel_pb2.Channel.Role.SECONDARY
-                    channels_to_write.append((disabled_channel.index, channel_name))
+                    pending_new_settings.append(chs)
                     existing_names.add(channel_name)
+                if len(pending_new_settings) > len(disabled_channels):
+                    self._raise_interface_error(
+                        "No free channels were found for all additions "
+                        f"(need {len(pending_new_settings)}, available {len(disabled_channels)})"
+                    )
+                for disabled_channel, new_settings in zip(
+                    disabled_channels, pending_new_settings
+                ):
```

Ruff (0.15.5)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)